lib/read_only/read_only_base.jinja

```diff
@@ -109,15 +109,28 @@ module read_only_base_mod
         procedure :: PreStart, PreEndDeclaration
         procedure :: PostStart
 
-        {# Declare the jinja created procedures #}
-        {# ------------------------------------ #}
-        {% for name, type, bits in ALL_TYPES %}
+        {# Collect the various procedures for the same generic interface -#}
+        {# ------------------------------------------------------------- -#}
+        {% set all_provides=[] -%}
+        {%- for name, type, bits in ALL_TYPES %}
         procedure :: ProvideScalar{{name}}
+          {{- all_provides.append("ProvideScalar"~name) or "" }}
           {% for dim in ALL_DIMS %}
         procedure :: ProvideArray{{dim}}d{{name}}
+            {{- all_provides.append("ProvideArray"~dim~"d"~name) or "" }}
           {% endfor %}
         {% endfor %}
 
+        {% set indent="            " %}
+
+        {% if GENERIC_PROVIDE %}
```

Comment:
Please add a comment explaining why this is only done if GENERIC_PROVIDE is set.

Reply:
Done.